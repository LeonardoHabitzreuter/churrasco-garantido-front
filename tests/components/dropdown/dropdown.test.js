import React from 'react'
import { shallow } from 'enzyme'
import Dropdown from '../../../src/components/dropdown'
import { MenuItem } from 'react-bootstrap'

describe('alert component', () => {
  test('should renders properly', () => {
    const wrapper = shallow(<Dropdown id='dropdownTest' />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should not break up when the items are an empty array', () => {
    shallow(<Dropdown id='dropdownTest' items={[]} />)
  })

  test('should renders the items', () => {
    const wrapper = shallow(<Dropdown id='dropdownTest' items={['Test1', 'Test2']} />)
    const items = wrapper.find(MenuItem)
    expect(items).toHaveLength(2)
    expect(items.map(item => item.children().text())).toEqual(expect.arrayContaining(['Test1', 'Test2']))
  })

  test('should call onChange passing the item index when the item is selected', () => {
    const callback = jest.fn()
    const wrapper = shallow(<Dropdown id='dropdownTest' items={['Test1', 'Test2']} onChange={callback} />)
    const test2 = wrapper.find(MenuItem).filterWhere(item => item.children().text() === 'Test2')
    test2.simulate('select')
    expect(callback).toHaveBeenCalledWith(1)
  })
})
