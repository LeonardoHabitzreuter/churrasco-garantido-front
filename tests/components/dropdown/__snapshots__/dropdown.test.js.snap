// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert component should renders properly 1`] = `
<DropdownButton
  bsStyle="default"
  id="dropdownTest"
  title=""
/>
`;
